rpcserver.go

```diff
@@ -3312,12 +3314,56 @@ func (r *rpcServer) BurnAsset(ctx context.Context,
 		}
 	}
 
+	// At this point everything completed correctly, so we log this burn in
+	// our db.
+	err = r.cfg.AssetStore.InsertBurn(
+		ctx, parcel.AnchorTxHash, in.Note, assetID[:],
+		serializedGroupKey, in.AmountToBurn,
+	)
+	if err != nil {
+		return nil, err
+	}
+
 	return &taprpc.BurnAssetResponse{
 		BurnTransfer: parcel,
 		BurnProof:    burnProof,
 	}, nil
 }
 
+// ListBurns returns a list of burnt assets. Some filters may be defined in the
+// request to return more specific results.
+func (r *rpcServer) ListBurns(ctx context.Context,
+	in *taprpc.ListBurnsRequest) (*taprpc.ListBurnsResponse, error) {
+
+	burns, err := r.cfg.AssetStore.QueryBurns(
+		ctx, sqlc.QueryBurnsParams{
+			AssetID:    in.AssetId,
+			GroupKey:   in.TweakedGroupKey,
+			AnchorTxid: in.AnchorTxid,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	rpcBurns := fn.Map(burns, marshalRpcBurn)
+
+	return &taprpc.ListBurnsResponse{
+		Burns: rpcBurns,
```

Comment:
Re other comment, I think it would be useful to include the full transfer here, also maybe the raw tx itself?

Reply:
a bit overblown? I'd expect the common usage of this call being about accounting, where only assetid/groupid and amount matters

if for any other reason (perhaps debugging related) the anchor tx / transfer is needed you could grab the anchor_txid and query them directly from the respective RPCs

Reply:
That's reasonable, if we need to expose it later, then it's on disk already i the `chain_txns` table.